### ISSUE TITLE
First and last name fields cannot be re-added to form once removed

### ISSUE THREAD

**Issue by pippinsplugins** [easydigitaldownloads/edd-checkout-fields#151]
If the first and / or last name fields are removed from the form, it is _impossible_ to add them back.

- The UI does nothing when trying to re-add them
- Resetting the form does not work
- Deleting DB options and re-creating does not work
- Absolutely nothing brings the first and last name fields back
- Exporting a resetted form shows the first and last name fields in the DB but UI does not show them

**Comment by SDavisMedia** [comment]
I think this created a new problem. Here's a ticket: https://secure.helpscout.net/conversation/507484924/74307/?folderId=180504

I was able to replicate by simply being on 2.1.1, pulling down `master` (2.1.2), and refreshing the Checkout Form admin page. My First Name and Last Name fields were removed from the form and I was hit with these notices:

```
[17-Jan-2018 18:08:58 UTC] PHP Notice:  Inserting a custom field without using CFM Fields API is <strong>deprecated</strong> since Easy Digital Downloads Checkout Fields Manager version 2.0. in /Applications/MAMP/htdocs/edd-dev/wp-content/plugins/edd-checkout-fields/classes/misc-functions.php on line 262
[17-Jan-2018 18:08:58 UTC] PHP Notice:   in /Applications/MAMP/htdocs/edd-dev/wp-content/plugins/edd-checkout-fields/classes/misc-functions.php on line 263
[17-Jan-2018 18:08:58 UTC] PHP Notice:  Inserting a custom field without using CFM Fields API is <strong>deprecated</strong> since Easy Digital Downloads Checkout Fields Manager version 2.0. in /Applications/MAMP/htdocs/edd-dev/wp-content/plugins/edd-checkout-fields/classes/misc-functions.php on line 262
[17-Jan-2018 18:08:58 UTC] PHP Notice:   in /Applications/MAMP/htdocs/edd-dev/wp-content/plugins/edd-checkout-fields/classes/misc-functions.php on line 263
```

I've rolled the released version back to 2.1.1 so we can look into what's happening here.

**Comment by SDavisMedia** [comment]
Been digging into this but I really need to get back to support. So gotta shelf it for now.

Two things I'm having issues with:

1. This change seems to be a problem: https://github.com/easydigitaldownloads/edd-checkout-fields/pull/152/commits/7d9d5a5e17316aeb480ab0c5c82d015554b307db#diff-43933cee3aaa47b0e349578c94fad120 I can confirm that when those keys aren't changed in the updated version, they are not lost during the update.

2. But with that ^^ reverted, as @pippinsplugins mentioned, the UI does not respond when you try to add the First Name or Last Name field to the form.

So basically there's an undesirable bug whether the keys get changed or not. Right now, though, it's looking like changing the keys is going to affect users the most.

**Comment by pippinsplugins** [comment]
@SDavisMedia ah crud. I should have added an upgrade routine that converts existing fields to the new keys.

Opening new issue and working on it.
